src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/AddUsingsCodeActionResolver.cs

```diff
@@ -0,0 +1,239 @@
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Components;
+using Microsoft.AspNetCore.Razor.Language.Extensions;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions
+{
+    class AddUsingsCodeActionResolver : RazorCodeActionResolver
+    {
+        private readonly ForegroundDispatcher _foregroundDispatcher;
+        private readonly DocumentResolver _documentResolver;
+
+        public AddUsingsCodeActionResolver(ForegroundDispatcher foregroundDispatcher, DocumentResolver documentResolver)
+        {
+            _foregroundDispatcher = foregroundDispatcher ?? throw new ArgumentNullException(nameof(foregroundDispatcher));
+            _documentResolver = documentResolver ?? throw new ArgumentNullException(nameof(documentResolver));
+        }
+
+        public override string Action => LanguageServerConstants.CodeActions.AddUsings;
+
+        public override async Task<WorkspaceEdit> ResolveAsync(JObject data, CancellationToken cancellationToken)
+        {
+            var actionParams = data.ToObject<AddUsingsCodeActionParams>();
+            var path = actionParams.Uri.GetAbsoluteOrUNCPath();
+
+            var document = await Task.Factory.StartNew(() =>
+            {
+                _documentResolver.TryResolveDocument(path, out var documentSnapshot);
+                return documentSnapshot;
+            }, cancellationToken, TaskCreationOptions.None, _foregroundDispatcher.ForegroundScheduler).ConfigureAwait(false);
+            if (document is null)
+            {
+                return null;
+            }
+
+            var text = await document.GetTextAsync().ConfigureAwait(false);
+            if (text is null)
+            {
+                return null;
+            }
+
+            var codeDocument = await document.GetGeneratedOutputAsync().ConfigureAwait(false);
+            if (codeDocument.IsUnsupported())
+            {
+                return null;
+            }
+
+            if (!FileKinds.IsComponent(codeDocument.GetFileKind()))
+            {
+                return null;
+            }
+
+            var codeDocumentIdentifier = new VersionedTextDocumentIdentifier() { Uri = actionParams.Uri, Version = 0 };
+            var documentChanges = new List<WorkspaceEditDocumentChange>();
+
+            var namespaceList = actionParams.Namespaces;
+
+            var usingDirectives = FindUsingDirectives(codeDocument);
+            if (usingDirectives.Count > 0)
+            {
+                documentChanges.Add(GenerateUsingEditsInterpolated(codeDocument, codeDocumentIdentifier, actionParams.Namespaces, usingDirectives));
```

Comment:
Break `GenerateUsingEditsInterpolated` into its own local variable for debuggability